designs/helm-charts/local-bind-volumes.md

```diff
@@ -0,0 +1,119 @@
+# Title: OpenEBS Enhancement Proposal for Mayastor Local Feature
+
+### OEP Number: <To be assigned>
+### Title: Reintroduction of Local Feature in Mayastor for Single-Replica Volumes
+### Authors: @todeb
+### Status: Proposed
+### Created: 2024-11-07
+### Last Updated: 2024-11-07
+### Target Release: ?
+
+---
+
+## Table of Contents
+
+1. [Overview](#overview)
+2. [Motivation](#motivation)
+3. [Goals](#goals)
+4. [Non-Goals](#non-goals)
+5. [Proposal](#proposal)
+6. [User Stories](#user-stories)
+7. [Implementation Details](#implementation-details)
+8. [Drawbacks](#drawbacks)
+9. [Alternatives](#alternatives)
+10. [Unresolved Questions](#unresolved-questions)
+
+---
+
+## 1. Overview
+
+This proposal reintroduces a "local" feature for the Mayastor storage engine in OpenEBS, designed specifically for single-replica volumes. This feature ensures that:
+
+1. A Mayastor Persistent Volume (PV) is created on the same node where the consuming pod is initially scheduled (using `volumeBindingMode: WaitForFirstConsumer`).
+2. Pods are prevented from rescheduling onto nodes where their Mayastor PVs are not present, maintaining strict local attachment to the original node.
+3. Users can configure the local behavior using a `local` flag, which can be set or unset via `kubectl-mayastor` commands.
+
+By implementing these features, the local configuration improves performance and resilience in environments without dedicated storage nodes.
+
+## 2. Motivation
+
+In clusters that do not have dedicated storage nodes, keeping application pods on the same node as their Mayastor volumes is essential. While `volumeBindingMode: WaitForFirstConsumer` is intended to ensure that volumes are created on the same node as their consuming pods, it does not always work as expected with Mayastor volumes. Reintroducing the `local` feature with this proposal enforces both local provisioning and node-level affinity for single-replica volumes, thereby avoiding cross-node data access and potential latency or availability issues.
+
+## 3. Goals
+
+- Implement the local feature for single-replica volumes in Mayastor, ensuring that PVs are created on the same node as their consuming pods when `volumeBindingMode: WaitForFirstConsumer` is used.
+- Prevent pods from being rescheduled to nodes where the associated volume does not exist.
+- Allow users to set or unset the `local` flag for a volume through `kubectl-mayastor`, providing flexibility and administrative control.
+
+## 4. Non-Goals
+
+- This proposal does not implement the local feature for multi-replica volumes, as cross-node replication complicates strict local binding.
+- The local feature is specific to Mayastor and does not apply to other OpenEBS engines, such as Jiva or cStor.
```

Comment:
I think you can leave Jiva and cStor out, as they're no longer OpenEBS engines (they're archived)